src/topology/metric_space/basic.lean

```diff
@@ -110,6 +119,43 @@ pseudo_metric_space.to_uniform_space
 @[priority 200] -- see Note [lower instance priority]
 instance pseudo_metric_space.to_has_edist : has_edist α := ⟨pseudo_metric_space.edist⟩
 
+/-- Construct a pseudo-metric space structure whose underlying topological space structure
+(definitionaly) agrees which a pre-existing topology which is compatible with a given distance
+function. -/
+def pseudo_metric_space.of_metrizable {α : Type*} [topological_space α] (dist : α → α → ℝ)
+  (dist_self : ∀ x : α, dist x x = 0)
+  (dist_comm : ∀ x y : α, dist x y = dist y x)
+  (dist_triangle : ∀ x y z : α, dist x z ≤ dist x y + dist y z)
+  (H : ∀ s : set α, is_open s ↔ ∀ x ∈ s, ∃ ε > 0, ∀ y, dist x y < ε → y ∈ s) :
+pseudo_metric_space α :=
+{ dist := dist,
+  dist_self := λ x, dist_self _,
+  dist_comm := λ x y, dist_comm _ _,
+  dist_triangle := λ x y z, dist_triangle _ _  _,
```

Comment:
Untested:
```suggestion
  dist_self := dist_self,
  dist_comm := dist_comm,
  dist_triangle := dist_triangle,
```

Reply:
Nice catch. I copy pasted this from a slightly different context with different binder types.